dojson/contrib/marc21/fields/ad01x09x.py

```diff
@@ -64,6 +65,7 @@ def national_bibliographic_agency_control_number(self, key, value):
         'canceled_or_invalid_record_control_number': utils.force_list(
             value.get('z')
         ),
+        'national_bibliographic_agency': indicator_map1.get(key[3]),
```

Comment:
IMHO it's part of the `source` key (subfield 2). Maybe we should add default value: `Library and Archives Canada`

```python
'source': value.get('2', 'Library and Archives Canada' if key[3] == '7' else None)
```
Then we need to check the inverse function too.

--
http://www.loc.gov/marc/bibliographic/bd016.html